src/widget/weffectparameternamebase.cpp

```diff
@@ -66,6 +82,10 @@ void WEffectParameterNameBase::parameterUpdated() {
         m_text = kNoEffectString;
         setBaseTooltip(tr("No effect loaded."));
     }
+    optimumWidth = math_max(
+            optimumWidth,
+            metrics.size(0, m_text).width());
+    m_widthHint = optimumWidth;
```

Comment:
let's rename `optimumWidth` to `valueWidth` and 
```suggestion
    int optimumWidth = math_max(
            valueWidth,
            metrics.size(0, m_text).width());
    m_widthHint = optimumWidth + 2 * frameWidth();
```
and remove the debug ouput below

Reply:
with this the labels are okay, besides the fact that they are initially (before swapping parameters) expanded compared to 2.4